.github/workflows/check.yml

```diff
@@ -56,6 +56,23 @@ jobs:
 
       - name: Run Tests
         run: bundle exec fastlane ${{ matrix.lane }}
+
+      - name: Check Static Analyzer Output
+        id: analyzer-output
+        run: |
+          if [[ -z $(find ./derived_data -name "report-*.html") ]]; then
+            echo "Static Analyzer found no issues."
+          else
+            echo "Static Analyzer found some issues. HTML report will be available in Artifacts section. Failing build."
+            exit 1
+          fi
+      
+      - name: Static Analyzer Reports Uploading
+        if: ${{ failure() && steps.analyzer-output.outcome == 'failure' }}
```

Comment:
Possibly just my lack of knowledge about GitHub Actions, but is the `failure()` check not redundant here, given we're also checking for failure of a specific step?

Reply:
Yes, GitHub Actions are weird, all previous commits were basically fighting against this nonsense. https://github.community/t/workflow-skips-steps-besides-if-condition/115956

Reply:
😱 👍